repos/system_upgrade/el7toel8/actors/getinstalleddesktops/libraries/library.py

```diff
@@ -0,0 +1,17 @@
+from leapp.libraries.stdlib import api
+from leapp.libraries.common.rpms import has_package
+from leapp.models import InstalledRPM
+
+
+def get_installed_desktops():
+    api.current_logger().info("  Detecting desktop environments  ")
+    api.current_logger().info("==================================")
+
+    # Detect installed desktops by one of the base rpm packages
+    kde_desktop_installed = has_package(InstalledRPM, "plasma-workspace")
```

Comment:
Where can I get this package? It's not in the RHEL 7 base repo, and also not in Optional or Extras.

Also, it's not mentioned in the Package Evolution Service.

Reply:
I was trying two commands for KDE installation: `# yum groupinstall 'X Window System' 'KDE'` and `yum groupinstall kde-desktop`, the latter advised here https://access.redhat.com/solutions/371643. Both of them gave me the same result - installing the below set of package - **not including** the `plasma-workspace` one.
<details><summary>List of packages in the KDE pkg group:</summary>

```
==============================================================================================================================================================================================================================================
 Package                                                             Arch                                     Version                                                          Repository                                                Size
==============================================================================================================================================================================================================================================
Installing for group install "KDE":
 akonadi                                                             x86_64                                   1.9.2-4.el7                                                      rhel-7-server-eus-rpms                                   725 k
 akonadi-mysql                                                       x86_64                                   1.9.2-4.el7                                                      rhel-7-server-eus-rpms                                    16 k
 ark                                                                 x86_64                                   4.10.5-4.el7                                                     rhel-7-server-eus-rpms                                   277 k
 bluedevil                                                           x86_64                                   2.1-1.el7                                                        rhel-7-server-eus-rpms                                   378 k
 colord-kde                                                          x86_64                                   0.3.0-3.el7                                                      rhel-7-server-eus-rpms                                   131 k
 gwenview                                                            x86_64                                   4.10.5-5.el7                                                     rhel-7-server-eus-rpms                                   1.9 M
 kamera                                                              x86_64                                   4.10.5-5.el7                                                     rhel-7-server-eus-rpms                                    63 k
 kcalc                                                               x86_64                                   4.10.5-4.el7                                                     rhel-7-server-eus-rpms                                   175 k
 kcharselect                                                         x86_64                                   4.10.5-3.el7                                                     rhel-7-server-eus-rpms                                    93 k
 kcm-gtk                                                             x86_64                                   0.5.3-14.el7                                                     rhel-7-server-eus-rpms                                    54 k
 kcm_touchpad                                                        x86_64                                   0.3.1-11.el7                                                     rhel-7-server-eus-rpms                                    47 k
 kcolorchooser                                                       x86_64                                   4.10.5-3.el7                                                     rhel-7-server-eus-rpms                                    15 k
 kde-base-artwork                                                    noarch                                   4.10.5-2.el7                                                     rhel-7-server-eus-rpms                                   7.1 M
 kde-plasma-networkmanagement                                        x86_64                                   1:0.9.0.9-7.el7                                                  rhel-7-server-eus-rpms                                   538 k
 kde-print-manager                                                   x86_64                                   4.10.5-4.el7                                                     rhel-7-server-eus-rpms                                   272 k
 kde-settings-pulseaudio                                             noarch                                   19-23.8.el7                                                      rhel-7-server-eus-rpms                                    15 k
 kde-workspace                                                       x86_64                                   4.11.19-12.el7                                                   rhel-7-server-eus-rpms                                   9.6 M
 kdeaccessibility                                                    noarch                                   1:4.10.5-2.el7                                                   rhel-7-server-eus-rpms                                    14 k
 kdeadmin                                                            x86_64                                   7:4.10.5-4.el7                                                   rhel-7-server-eus-rpms                                   865 k
 kdegraphics-strigi-analyzer                                         x86_64                                   4.10.5-3.el7                                                     rhel-7-server-eus-rpms                                    28 k
 kdegraphics-thumbnailers                                            x86_64                                   4.10.5-3.el7                                                     rhel-7-server-eus-rpms                                    50 k
 kdenetwork-kdnssd                                                   x86_64                                   7:4.10.5-8.el7_0                                                 rhel-7-server-eus-rpms                                    39 k
 kdenetwork-kget                                                     x86_64                                   7:4.10.5-8.el7_0                                                 rhel-7-server-eus-rpms                                   1.3 M
 kdenetwork-krfb                                                     x86_64                                   7:4.10.5-8.el7_0                                                 rhel-7-server-eus-rpms                                   369 k
 kdepim                                                              x86_64                                   7:4.10.5-7.el7                                                   rhel-7-server-eus-rpms                                   5.6 M
 kdeplasma-addons                                                    x86_64                                   4.10.5-5.el7                                                     rhel-7-server-eus-rpms                                   2.4 M
 kdeutils-minimal                                                    noarch                                   6:4.10.5-3.el7                                                   rhel-7-server-eus-rpms                                    16 k
 kgpg                                                                x86_64                                   4.10.5-4.el7                                                     rhel-7-server-eus-rpms                                   962 k
 kmix                                                                x86_64                                   4.10.5-4.el7                                                     rhel-7-server-eus-rpms                                   574 k
 konsole                                                             x86_64                                   4.10.5-4.el7                                                     rhel-7-server-eus-rpms                                    78 k
 kruler                                                              x86_64                                   4.10.5-3.el7                                                     rhel-7-server-eus-rpms                                   134 k
 ksaneplugin                                                         x86_64                                   4.10.5-3.el7                                                     rhel-7-server-eus-rpms                                    26 k
 ksnapshot                                                           x86_64                                   4.10.5-3.el7                                                     rhel-7-server-eus-rpms                                   309 k
 ksshaskpass                                                         x86_64                                   0.5.3-7.el7                                                      rhel-7-server-eus-rpms                                    21 k
 ksysguard                                                           x86_64                                   4.11.19-12.el7                                                   rhel-7-server-eus-rpms                                   278 k
 kwallet                                                             x86_64                                   4.10.5-3.el7                                                     rhel-7-server-eus-rpms                                   351 k
 kwrite                                                              x86_64                                   4.10.5-6.el7                                                     rhel-7-server-eus-rpms                                    78 k
 okular                                                              x86_64                                   4.10.5-4.el7                                                     rhel-7-server-eus-rpms                                   413 k
 oxygen-gtk                                                          noarch                                   1.2.0-6.el7                                                      rhel-7-server-eus-rpms                                   5.0 k
 plasma-scriptengine-python                                          x86_64                                   4.11.19-12.el7                                                   rhel-7-server-eus-rpms                                    86 k
 sweeper                                                             x86_64                                   4.10.5-4.el7                                                     rhel-7-server-eus-rpms                                    97 k
 system-config-date                                                  noarch                                   1.10.6-3.el7                                                     rhel-7-server-eus-rpms                                   591 k
 xsettings-kde                                                       x86_64                                   0.12.3-7.el7                                                     rhel-7-server-eus-rpms                                    27 k
 xterm                                                               x86_64                                   295-3.el7                                                        rhel-7-server-eus-rpms                                   455 k
Installing:
 pinentry-qt                                                         x86_64                                   0.8.1-17.el7                                                     rhel-7-server-eus-rpms                                    66 k
Installing for dependencies:
 LibRaw                                                              x86_64                                   0.14.8-5.el7.20120830git98d925                                   rhel-7-server-eus-rpms                                   250 k
 PyQt4                                                               x86_64                                   4.10.1-13.el7                                                    rhel-7-server-eus-rpms                                   2.9 M
 ark-libs                                                            x86_64                                   4.10.5-4.el7                                                     rhel-7-server-eus-rpms                                   108 k
 boost-program-options                                               x86_64                                   1.53.0-27.el7                                                    rhel-7-server-eus-rpms                                   156 k
 compat-poppler022                                                   x86_64                                   0.22.5-4.el7                                                     rhel-7-server-eus-rpms                                   722 k
 compat-poppler022-qt                                                x86_64                                   0.22.5-4.el7                                                     rhel-7-server-eus-rpms                                   150 k
 cpp                                                                 x86_64                                   4.8.5-37.el7_6                                                   rhel-7-server-eus-rpms                                   6.0 M
 grantlee                                                            x86_64                                   0.3.0-4.el7                                                      rhel-7-server-eus-rpms                                   312 k
 gwenview-libs                                                       x86_64                                   4.10.5-5.el7                                                     rhel-7-server-eus-rpms                                   346 k
 kaccessible                                                         x86_64                                   4.10.5-3.el7                                                     rhel-7-server-eus-rpms                                    40 k
 kaccessible-libs                                                    x86_64                                   4.10.5-3.el7                                                     rhel-7-server-eus-rpms                                    20 k
 kcm_colors                                                          x86_64                                   4.11.19-12.el7                                                   rhel-7-server-eus-rpms                                   146 k
 kde-plasma-networkmanagement-libs                                   x86_64                                   1:0.9.0.9-7.el7                                                  rhel-7-server-eus-rpms                                   1.0 M
 kde-settings-ksplash                                                noarch                                   19-23.8.el7                                                      rhel-7-server-eus-rpms                                    16 k
 kde-settings-plasma                                                 noarch                                   19-23.8.el7                                                      rhel-7-server-eus-rpms                                    16 k
 kde-style-oxygen                                                    x86_64                                   4.11.19-12.el7                                                   rhel-7-server-eus-rpms                                   405 k
 kde-workspace-libs                                                  x86_64                                   4.11.19-12.el7                                                   rhel-7-server-eus-rpms                                   579 k
 kdenetwork-common                                                   noarch                                   7:4.10.5-8.el7_0                                                 rhel-7-server-eus-rpms                                    10 k
 kdenetwork-kget-libs                                                x86_64                                   7:4.10.5-8.el7_0                                                 rhel-7-server-eus-rpms                                   281 k
 kdenetwork-krfb-libs                                                x86_64                                   7:4.10.5-8.el7_0                                                 rhel-7-server-eus-rpms                                    16 k
 kdepim-libs                                                         x86_64                                   7:4.10.5-7.el7                                                   rhel-7-server-eus-rpms                                   3.2 M
 kdepim-runtime                                                      x86_64                                   1:4.10.5-3.el7                                                   rhel-7-server-eus-rpms                                   2.8 M
 kdepim-runtime-libs                                                 x86_64                                   1:4.10.5-3.el7                                                   rhel-7-server-eus-rpms                                   150 k
 kdepimlibs-akonadi                                                  x86_64                                   4.10.5-4.el7                                                     rhel-7-server-eus-rpms                                   1.6 M
 kdeplasma-addons-libs                                               x86_64                                   4.10.5-5.el7                                                     rhel-7-server-eus-rpms                                   369 k
 kdf                                                                 x86_64                                   4.10.5-3.el7                                                     rhel-7-server-eus-rpms                                   246 k
 kgreeter-plugins                                                    x86_64                                   4.11.19-12.el7                                                   rhel-7-server-eus-rpms                                    95 k
 khotkeys                                                            x86_64                                   4.11.19-12.el7                                                   rhel-7-server-eus-rpms                                   304 k
 khotkeys-libs                                                       x86_64                                   4.11.19-12.el7                                                   rhel-7-server-eus-rpms                                   150 k
 kinfocenter                                                         x86_64                                   4.11.19-12.el7                                                   rhel-7-server-eus-rpms                                   591 k
 kmag                                                                x86_64                                   4.10.5-3.el7                                                     rhel-7-server-eus-rpms                                   121 k
 kmenuedit                                                           x86_64                                   4.11.19-12.el7                                                   rhel-7-server-eus-rpms                                   361 k
 konsole-part                                                        x86_64                                   4.10.5-4.el7                                                     rhel-7-server-eus-rpms                                   393 k
 kscreen                                                             x86_64                                   1:1.0.1-9.el7                                                    rhel-7-server-eus-rpms                                   163 k
 ksysguard-libs                                                      x86_64                                   4.11.19-12.el7                                                   rhel-7-server-eus-rpms                                   240 k
 ksysguardd                                                          x86_64                                   4.11.19-12.el7                                                   rhel-7-server-eus-rpms                                   117 k
 ktimer                                                              x86_64                                   4.10.5-3.el7                                                     rhel-7-server-eus-rpms                                   162 k
 kwin                                                                x86_64                                   4.11.19-12.el7                                                   rhel-7-server-eus-rpms                                   2.3 M
 kwin-libs                                                           x86_64                                   4.11.19-12.el7                                                   rhel-7-server-eus-rpms                                   224 k
 libXaw                                                              x86_64                                   1.0.13-4.el7                                                     rhel-7-server-eus-rpms                                   192 k
 libbluedevil                                                        x86_64                                   2.1-1.el7                                                        rhel-7-server-eus-rpms                                    59 k
 libkdcraw                                                           x86_64                                   4.10.5-5.el7                                                     rhel-7-server-eus-rpms                                   120 k
 libkexiv2                                                           x86_64                                   4.10.5-4.el7                                                     rhel-7-server-eus-rpms                                   257 k
 libkipi                                                             x86_64                                   4.10.5-3.el7                                                     rhel-7-server-eus-rpms                                    99 k
 libksane                                                            x86_64                                   4.10.5-4.el7                                                     rhel-7-server-eus-rpms                                   104 k
 libkscreen                                                          x86_64                                   1:1.0.1-8.el7                                                    rhel-7-server-eus-rpms                                    75 k
 libkworkspace                                                       x86_64                                   4.11.19-12.el7                                                   rhel-7-server-eus-rpms                                   111 k
 libvncserver                                                        x86_64                                   0.9.9-13.el7_6                                                   rhel-7-server-eus-rpms                                   234 k
 lzo-minilzo                                                         x86_64                                   2.06-8.el7                                                       rhel-7-server-eus-rpms                                    15 k
 mariadb                                                             x86_64                                   1:5.5.60-1.el7_5                                                 rhel-7-server-eus-rpms                                   8.6 M
 mariadb-server                                                      x86_64                                   1:5.5.60-1.el7_5                                                 rhel-7-server-eus-rpms                                    11 M
 mesa-libGLU                                                         x86_64                                   9.0.0-4.el7                                                      rhel-7-server-eus-rpms                                   196 k
 okular-libs                                                         x86_64                                   4.10.5-4.el7                                                     rhel-7-server-eus-rpms                                   229 k
 okular-part                                                         x86_64                                   4.10.5-4.el7                                                     rhel-7-server-eus-rpms                                   839 k
 oxygen-gtk2                                                         x86_64                                   1.3.4-3.el7                                                      rhel-7-server-eus-rpms                                   431 k
 perl-DBD-MySQL                                                      x86_64                                   4.023-6.el7                                                      rhel-7-server-eus-rpms                                   140 k
 polkit-kde                                                          x86_64                                   0.99.1-4.20130311git.el7                                         rhel-7-server-eus-rpms                                    71 k
 pykde4                                                              x86_64                                   4.10.5-4.el7                                                     rhel-7-server-eus-rpms                                   4.2 M
 qimageblitz                                                         x86_64                                   0.0.6-7.el7                                                      rhel-7-server-eus-rpms                                    56 k
 qjson                                                               x86_64                                   0.8.1-4.el7                                                      rhel-7-server-eus-rpms                                    72 k
 qt-mysql                                                            x86_64                                   1:4.8.7-3.el7_6                                                  rhel-7-server-eus-rpms                                    32 k
 qt3                                                                 x86_64                                   3.3.8b-51.el7                                                    rhel-7-server-eus-rpms                                   3.5 M
 sip                                                                 x86_64                                   4.14.6-4.el7                                                     rhel-7-server-eus-rpms                                   122 k
 system-config-date-docs                                             noarch                                   1.0.11-4.el7                                                     rhel-7-server-eus-rpms                                   527 k
```

</details>

Reply:
Oh, I see the problem. I checked the packages for Fedora, not for RHEL. In Fedora, there really is package named 
`plasma-workspace`, but in RHEL it should be `kde-workspace` I think. I will check the differences and use the right package for RHEL.